pkg/aws/core.go

```diff
@@ -243,6 +247,20 @@ func (d *Driver) CreateMachine(ctx context.Context, req *driver.CreateMachineReq
 		}
 	}
 
+	for i, netIf := range providerSpec.NetworkInterfaces {
+		networkInterfaceID := runResult.Instances[0].NetworkInterfaces[i].NetworkInterfaceId
+		if netIf.Ipv6PrefixCount != nil {
+			input := &ec2.AssignIpv6AddressesInput{
+				NetworkInterfaceId: networkInterfaceID,
+				Ipv6PrefixCount:    aws.Int64(*netIf.Ipv6PrefixCount),
+			}
+			_, err = svc.AssignIpv6Addresses(input)
+			if err != nil {
```

Comment:
shouldn't we also check whether the networkInterface has IPv6 IPs assigned before getting ipv6 delegated prefixes for it ? or are you planning to have ipv6 delegated prefixes for nics with ipv4 addresses also?

Reply:
The `Ipv6PrefixCount` and `Ipv6AddressCount` can't be set in one request. However, having an Ipv6 address is not a prerequisite for getting a prefix.
We don't plan to have IPv6 prefixes for instances with only IPv4 addresses. Actually, I can't imagine how that would make sense. However, from AWS side it is still possible.
We wouldn't enable that combination in a `NodeClass`. 
Do we really need to validate that here?

Reply:
The concern is not from Gardener consumption side , but for users consuming mcm-aws directly. If they provide multiple nics specification in the providerSpec , like in order  { ipv4 , ipv6 with `ipv6prefixCount` set , ipv6 with `ipv6AddressCount` set, ipv4} and then in this for loop, `ipv6prefixCount` might get set up for an `ipv4` nic (depends on how `runResult.Instances[0].NetworkInterfaces[]` are arranged)... 

so we should deal with this either via validation or someway else.

Reply:
I was assuming that `DeviceIndex` would be the index in that array. However, I didn't find docs that explicitly state that this is really the case. Now, I explicitly look for a matching `DeviceIndex`.